ledger/eval.go

```diff
@@ -697,14 +656,100 @@ func (eval *BlockEvaluator) GenerateBlock() (*ValidatedBlock, error) {
 	return &vb, nil
 }
 
+type evalTxValidator struct {
+	txcache          VerifiedTxnCache
+	block            bookkeeping.Block
+	proto            config.ConsensusParams
+	verificationPool execpool.BacklogPool
+
+	ctx      context.Context
+	cf       context.CancelFunc
+	txgroups chan []transactions.SignedTxnWithAD
+	done     chan error
+}
+
+func (tv *evalTxValidator) run() {
+	for txgroup := range tv.txgroups {
+		select {
+		case <-tv.ctx.Done():
+			tv.done <- tv.ctx.Err()
+			tv.cf()
+			close(tv.done)
+			return
+		default:
+		}
+		groupNoAD := make([]transactions.SignedTxn, len(txgroup))
+		for i := range txgroup {
+			groupNoAD[i] = txgroup[i].SignedTxn
+		}
+		ctxs := verify.PrepareContexts(groupNoAD, tv.block.BlockHeader)
+
+		for gi, tx := range txgroup {
+			err := validateTransaction(tx.SignedTxn, tv.block, tv.proto, tv.txcache, ctxs[gi], tv.verificationPool)
```

Comment:
is it intentional that this is not checking in parallel? (since it waits in lockstep for each transaction to verify)

Reply:
It's not intentional. It's marginally better than doing it inline in BlockEvaluator (which processes all txns serially). So at least validate and ledger accounting are running in two parallel threads. A future update could run more validates in parallel if that's a useful optimization?